### ISSUE TITLE
Неэффективное использование словаря

### ISSUE THREAD

**Issue by DanaZL** [AZuev1986/lp_lesson2#4]
https://github.com/AZuev1986/lp_lesson2/blob/f0704e2ed17356832bbaa5360abf25a74563252f/while.py#L12

Зачем делать проход по всем ключам в словаре, если мы умеем быстро узнавать есть ли в нем ключ user_say?

**Comment by AntZuev** [comment]
да, ты права. Просто я почему-то думал, что  dict.get(key) тоже проходит по всему словарю, когда возвращает значение

**Comment by DanaZL** [comment]
В этом как раз и прелесть словаря - он быстро возвращает значение по ключу, независимо от размера словаря. Он использует хэширование - можешь почитать поподробнее, если интересно :)

**Comment by DanaZL** [comment]
Посмотрела исправление, когда мы пишем строчку:
if user_say in  DICT_ANSWER.keys():
мы опять итерируемся по все ключам в словаре (так как if делает проверку по всему интератору DICT_ANSWER.keys()), хотя для проверки есть ли значение в словаре это абсолютно необязательно

**Comment by AntZuev** [comment]
Исправил.
